### PR TITLE
[WIP]Refactor modules using java stream api's

### PR DESCRIPTION
**Pull request by Akanksha-kedia**
## Description
Refactor modules using java stream api's

## Motivation and Context
Refactor modules using java stream api's as unnecessary return true, false and also if else reduces code quality.

## Impact
No impact

## Test Plan
UT for modules where changes done.

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java [hunk 1]
```diff
@@ -641,11 +641,7 @@ protected Boolean visitSimpleCaseExpression(SimpleCaseExpression node, Void cont
                 }
             }
 
-            if (node.getDefaultValue().isPresent() && !process(node.getDefaultValue().get(), context)) {
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java:644]: I find the prior version more readable and understandable.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java [hunk 2]
```diff
@@ -343,11 +343,7 @@ private boolean checkAllEquatableTypes(AggregateInfo aggregateInfo)
                 }
             }
 
-            if (!aggregateInfo.getMask().getType().isComparable()) {
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java:346]: I find the prior version more readable and understandable.

File: presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainOrcPredicate.java [hunk 3]
```diff
@@ -135,10 +135,7 @@ private boolean columnOverlaps(ColumnReference<C> columnReference, Domain predic
         }
 
         // if none of the discrete predicate values are found in the bloom filter, there is no overlap and the section should be skipped
-        if (discreteValues.get().stream().noneMatch(value -> checkInBloomFilter(bloomFilter, value, stripeDomain.getType()))) {
```
Comment (tdcmeehan) [inline presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainOrcPredicate.java:138]: I find the prior version more readable and understandable.

File: presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java [hunk 4]
```diff
@@ -115,9 +117,20 @@ public int countBitsSet(int items)
...
+                            return nextBit() ? 1 : 0;
+                        }
+                        catch (IOException e) {
+                            throw new UncheckedIOException(e);
+                        }
+                    })
+                    .sum();
+        }
+        catch (UncheckedIOException e) {
+            e.printStackTrace();
```
Comment (tdcmeehan) [inline presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java:133]: We cannot print a stack trace to console and drop the exception.

File: presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java [hunk 5]
```diff
@@ -397,8 +410,20 @@ public int getUnsetBits(int batchSize)
...
```
Comment (tdcmeehan) [inline presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java:426]: We cannot print the stack trace to console and drop the exception, it must be propogated.

File: presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java [hunk 6]
```diff
@@ -1469,7 +1469,7 @@ private static List<List<?>> pruneValues(List<Type> types, List<List<?>> values,
     public static void assertColumnValueEquals(Type type, Object actual, Object expected)
     {
         if (actual == null) {
-            assertEquals(actual, expected);
```
Comment (tdcmeehan) [inline presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java:1472]: Prior code is better.

File: presto-orc/src/test/java/com/facebook/presto/orc/TestingOrcPredicate.java [hunk 7]
```diff
@@ -241,11 +241,7 @@ private void assertChunkStats(List<T> chunk, ColumnStatistics columnStatistics)
         protected boolean chunkMatchesStats(List<T> chunk, ColumnStatistics columnStatistics)
         {
             // verify non null count
-            if (columnStatistics.getNumberOfValues() != Iterables.size(filter(chunk, notNull()))) {
```
Comment (tdcmeehan) [inline presto-orc/src/test/java/com/facebook/presto/orc/TestingOrcPredicate.java:244]: I find the prior version more readable and understandable.

File: presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/TupleDomainParquetPredicate.java [hunk 8]
```diff
@@ -542,11 +542,7 @@ static class ParquetUserDefinedPredicateTupleDomain<T extends Comparable<T>>
         @Override
         public boolean keep(T value)
         {
-            if (value == null && !columnDomain.isNullAllowed()) {
```
Comment (tdcmeehan) [inline presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/TupleDomainParquetPredicate.java:545]: I find the prior version more readable and understandable.

**Review by tdcmeehan — CHANGES_REQUESTED** [review]

File: presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java (hunk 4 above)
Comment (sdruzkin) [reply presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java:133]: Are you making a change in this file just for the sake of the making a change?

Please don't do it. You are not only making it unreadable, you are also degrading performance by this change.

**Comment by Akanksha-kedia** [comment]
its in work in progress @tdcmeehan @sdruzkin i was trying. but thanks for the review.

**Pull request closed without merging**